diff -ur Carla/resources/resources.qrc Carla.my17/resources/resources.qrc
--- Carla/resources/resources.qrc	2025-03-07 22:50:26.241329109 +0300
+++ Carla.my17/resources/resources.qrc	2025-03-25 19:48:07.047651719 +0300
@@ -3,9 +3,15 @@
         <file>16x16/carla.png</file>
         <file>16x16/carla-control.png</file>
 
+        <file>16x16/add-from-favorites.svgz</file>
+        <file>16x16/add-jack.svgz</file>
         <file>16x16/application-exit.svgz</file>
         <file>16x16/arrow-right.svgz</file>
+        <file>16x16/audio-volume-medium.svgz</file>
+        <file>16x16/audio-volume-muted.svgz</file>
+        <file>16x16/balance.svgz</file>
         <file>16x16/bookmarks.svgz</file>
+        <file>16x16/compact.svgz</file>
         <file>16x16/configure.svgz</file>
         <file>16x16/dialog-cancel.svgz</file>
         <file>16x16/dialog-error.svgz</file>
@@ -16,9 +22,11 @@
         <file>16x16/document-open.svgz</file>
         <file>16x16/document-save.svgz</file>
         <file>16x16/document-save-as.svgz</file>
+        <file>16x16/dry.svgz</file>
         <file>16x16/edit-clear.svgz</file>
         <file>16x16/edit-delete.svgz</file>
         <file>16x16/edit-rename.svgz</file>
+        <file>16x16/emblem-favorite.svgz</file>
         <file>16x16/list-add.svgz</file>
         <file>16x16/list-remove.svgz</file>
         <file>16x16/media-playback-pause.svgz</file>
@@ -27,8 +35,14 @@
         <file>16x16/media-seek-backward.svgz</file>
         <file>16x16/media-seek-forward.svgz</file>
         <file>16x16/network-connect.svgz</file>
+        <file>16x16/restore.svgz</file>
+        <file>16x16/skin.svgz</file>
+        <file>16x16/system-shutdown.svgz</file>
+        <file>16x16/system-turnon.svgz</file>
         <file>16x16/view-refresh.svgz</file>
+        <file>16x16/view-refresh-purple.svgz</file>
         <file>16x16/view-sort-ascending.svgz</file>
+        <file>16x16/wet.svgz</file>
         <file>16x16/window-close.svgz</file>
         <file>16x16/zoom-fit-best.svgz</file>
         <file>16x16/zoom-in.svgz</file>
diff -ur Carla/resources/ui/carla_edit.ui Carla.my17/resources/ui/carla_edit.ui
--- Carla/resources/ui/carla_edit.ui	2025-03-07 22:50:26.245685491 +0300
+++ Carla.my17/resources/ui/carla_edit.ui	2025-03-25 19:48:07.049092604 +0300
@@ -108,17 +108,17 @@
              </spacer>
             </item>
             <item>
-             <widget class="ScalableDial" name="dial_drywet">
+             <widget class="QWidget" name="dial_drywet">
               <property name="minimumSize">
                <size>
-                <width>34</width>
-                <height>34</height>
+                <width>32</width>
+                <height>48</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
-                <width>34</width>
-                <height>34</height>
+                <width>32</width>
+                <height>48</height>
                </size>
               </property>
               <property name="contextMenuPolicy">
@@ -130,17 +130,17 @@
              </widget>
             </item>
             <item>
-             <widget class="ScalableDial" name="dial_vol">
+             <widget class="QWidget" name="dial_vol">
               <property name="minimumSize">
                <size>
-                <width>34</width>
-                <height>34</height>
+                <width>32</width>
+                <height>48</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
-                <width>34</width>
-                <height>34</height>
+                <width>32</width>
+                <height>48</height>
                </size>
               </property>
               <property name="contextMenuPolicy">
@@ -152,148 +152,92 @@
              </widget>
             </item>
             <item>
-             <widget class="QStackedWidget" name="stackedWidget">
+              <widget class="QWidget" name="dial_b_left">
+              <property name="minimumSize">
+               <size>
+                <width>26</width>
+                <height>40</height>
+               </size>
+              </property>
+              <property name="maximumSize">
+               <size>
+                <width>26</width>
+                <height>40</height>
+               </size>
+              </property>
+              <property name="contextMenuPolicy">
+               <enum>Qt::CustomContextMenu</enum>
+              </property>
+              <property name="statusTip">
+               <string>Balance Left (0%)</string>
+              </property>
+             </widget>
+            </item>
+            <item>
+              <widget class="QWidget" name="dial_b_right">
+              <property name="minimumSize">
+               <size>
+                <width>26</width>
+                <height>40</height>
+               </size>
+              </property>
+              <property name="maximumSize">
+               <size>
+                <width>26</width>
+                <height>40</height>
+               </size>
+              </property>
+              <property name="contextMenuPolicy">
+               <enum>Qt::CustomContextMenu</enum>
+              </property>
+              <property name="statusTip">
+               <string>Balance Right (0%)</string>
+              </property>
+             </widget>
+            </item>
+            <item>
+              <widget class="QWidget" name="dial_pan">
+              <property name="minimumSize">
+               <size>
+                <width>32</width>
+                <height>48</height>
+               </size>
+              </property>
               <property name="maximumSize">
                <size>
-                <width>16777215</width>
-                <height>42</height>
+                <width>32</width>
+                <height>48</height>
                </size>
               </property>
-              <property name="lineWidth">
-               <number>0</number>
+              <property name="contextMenuPolicy">
+               <enum>Qt::CustomContextMenu</enum>
+              </property>
+              <property name="statusTip">
+               <string>Left-Right (0%)</string>
               </property>
-              <property name="currentIndex">
-               <number>0</number>
-              </property>
-              <widget class="QWidget" name="page_bal">
-               <layout class="QHBoxLayout" name="horizontalLayout_6">
-                <property name="spacing">
-                 <number>0</number>
-                </property>
-                <property name="leftMargin">
-                 <number>0</number>
-                </property>
-                <property name="topMargin">
-                 <number>0</number>
-                </property>
-                <property name="rightMargin">
-                 <number>0</number>
-                </property>
-                <property name="bottomMargin">
-                 <number>0</number>
-                </property>
-                <item>
-                 <widget class="ScalableDial" name="dial_b_left">
-                  <property name="minimumSize">
-                   <size>
-                    <width>26</width>
-                    <height>26</height>
-                   </size>
-                  </property>
-                  <property name="maximumSize">
-                   <size>
-                    <width>26</width>
-                    <height>26</height>
-                   </size>
-                  </property>
-                  <property name="contextMenuPolicy">
-                   <enum>Qt::CustomContextMenu</enum>
-                  </property>
-                  <property name="statusTip">
-                   <string>Balance Left (0%)</string>
-                  </property>
-                 </widget>
-                </item>
-                <item>
-                 <widget class="ScalableDial" name="dial_b_right">
-                  <property name="minimumSize">
-                   <size>
-                    <width>26</width>
-                    <height>26</height>
-                   </size>
-                  </property>
-                  <property name="maximumSize">
-                   <size>
-                    <width>26</width>
-                    <height>26</height>
-                   </size>
-                  </property>
-                  <property name="contextMenuPolicy">
-                   <enum>Qt::CustomContextMenu</enum>
-                  </property>
-                  <property name="statusTip">
-                   <string>Balance Right (0%)</string>
-                  </property>
-                 </widget>
-                </item>
-               </layout>
-              </widget>
-              <widget class="QWidget" name="page_pan">
-               <layout class="QHBoxLayout" name="horizontalLayout_7">
-                <property name="spacing">
-                 <number>0</number>
-                </property>
-                <property name="leftMargin">
-                 <number>0</number>
-                </property>
-                <property name="topMargin">
-                 <number>0</number>
-                </property>
-                <property name="rightMargin">
-                 <number>0</number>
-                </property>
-                <property name="bottomMargin">
-                 <number>0</number>
-                </property>
-                <item>
-                 <widget class="ScalableDial" name="dial_pan">
-                  <property name="minimumSize">
-                   <size>
-                    <width>26</width>
-                    <height>26</height>
-                   </size>
-                  </property>
-                  <property name="maximumSize">
-                   <size>
-                    <width>26</width>
-                    <height>26</height>
-                   </size>
-                  </property>
-                  <property name="contextMenuPolicy">
-                   <enum>Qt::CustomContextMenu</enum>
-                  </property>
-                  <property name="statusTip">
-                   <string>Balance Right (0%)</string>
-                  </property>
-                 </widget>
-                </item>
-               </layout>
-              </widget>
              </widget>
             </item>
             <item>
-             <layout class="QVBoxLayout" name="verticalLayout_2">
-              <property name="spacing">
-               <number>0</number>
-              </property>
-              <item>
-               <widget class="QRadioButton" name="rb_balance">
-                <property name="text">
-                 <string>Use Balance</string>
-                </property>
-                <property name="checked">
-                 <bool>true</bool>
-                </property>
-               </widget>
-              </item>
-              <item>
-               <widget class="QRadioButton" name="rb_pan">
-                <property name="text">
-                 <string>Use Panning</string>
-                </property>
-               </widget>
-              </item>
-             </layout>
+              <widget class="QWidget" name="dial_forth">
+              <property name="minimumSize">
+               <size>
+                <width>32</width>
+                <height>48</height>
+               </size>
+              </property>
+              <property name="maximumSize">
+               <size>
+                <width>32</width>
+                <height>48</height>
+               </size>
+              </property>
+              <property name="contextMenuPolicy">
+               <enum>Qt::CustomContextMenu</enum>
+              </property>
+              <property name="statusTip">
+               <string>Front-Rear (0%)</string>
+              </property>
+             </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_4">
@@ -310,6 +254,21 @@
             </item>
            </layout>
           </item>
+          <item>
+           <widget class="QLabel" name="label_ctrl_channel_2">
+            <property name="font">
+             <font>
+              <italic>true</italic>
+             </font>
+            </property>
+            <property name="text">
+             <string>&#9888; L, R are special mixing type.</string>
+            </property>
+            <property name="alignment">
+             <set>Qt::AlignCenter</set>
+            </property>
+           </widget>
+          </item>
          </layout>
         </widget>
        </item>
@@ -876,13 +835,6 @@
    </item>
   </layout>
  </widget>
- <customwidgets>
-  <customwidget>
-   <class>ScalableDial</class>
-   <extends>QDial</extends>
-   <header>widgets/scalabledial.h</header>
-  </customwidget>
- </customwidgets>
  <resources>
   <include location="../resources.qrc"/>
  </resources>
diff -ur Carla/resources/ui/carla_host.ui Carla.my17/resources/ui/carla_host.ui
--- Carla/resources/ui/carla_host.ui	2025-03-07 22:50:26.245685491 +0300
+++ Carla.my17/resources/ui/carla_host.ui	2025-03-25 19:48:07.050186749 +0300
@@ -474,6 +474,7 @@
     <addaction name="act_file_refresh"/>
     <addaction name="act_file_new"/>
     <addaction name="act_file_open"/>
+    <addaction name="act_file_reload"/>
     <addaction name="act_file_save"/>
     <addaction name="act_file_save_as"/>
     <addaction name="separator"/>
@@ -508,6 +509,7 @@
      <addaction name="separator"/>
      <addaction name="act_plugins_center"/>
      <addaction name="separator"/>
+     <addaction name="act_plugins_change_skin"/>
      <addaction name="act_plugins_compact"/>
      <addaction name="act_plugins_expand"/>
     </widget>
@@ -548,6 +550,7 @@
      <string>&amp;Settings</string>
     </property>
     <addaction name="act_settings_show_toolbar"/>
+    <addaction name="act_settings_show_toolbar_text"/>
     <addaction name="act_settings_show_meters"/>
     <addaction name="act_settings_show_keyboard"/>
     <addaction name="act_settings_show_side_panel"/>
@@ -589,6 +592,7 @@
    </attribute>
    <addaction name="act_file_new"/>
    <addaction name="act_file_open"/>
+   <addaction name="act_file_reload"/>
    <addaction name="act_file_save"/>
    <addaction name="act_file_save_as"/>
    <addaction name="act_file_connect"/>
@@ -600,6 +604,17 @@
    <addaction name="act_engine_panic"/>
    <addaction name="separator"/>
    <addaction name="act_settings_configure"/>
+   <addaction name="separator"/>
+   <addaction name="act_plugins_enable"/>
+   <addaction name="act_plugins_disable"/>
+   <addaction name="act_plugins_bypass"/>
+   <addaction name="act_plugins_wet100"/>
+   <addaction name="act_plugins_mute"/>
+   <addaction name="act_plugins_volume100"/>
+   <addaction name="act_plugins_center"/>
+   <addaction name="act_plugins_change_skin"/>
+   <addaction name="act_plugins_compact"/>
+   <addaction name="act_plugins_expand"/>
   </widget>
   <widget class="QDockWidget" name="dockWidget">
    <property name="sizePolicy">
@@ -1111,6 +1126,30 @@
     <enum>QAction::NoRole</enum>
    </property>
   </action>
+  <action name="act_file_reload">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/view-refresh-purple.svgz</normaloff>:/16x16/view-refresh-purple.svgz</iconset>
+   </property>
+   <property name="text">
+    <string>&amp;Reload (!)</string>
+   </property>
+   <property name="iconText">
+    <string>Reload (!)</string>
+   </property>
+   <property name="toolTip">
+    <string>Reload file. CAUTION, non-saved changes will be LOST!</string>
+   </property>
+   <property name="shortcut">
+    <string>Ctrl+Shift+R</string>
+   </property>
+   <property name="visible">
+    <bool>false</bool>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
   <action name="act_file_save">
    <property name="icon">
     <iconset resource="../resources.qrc">
@@ -1220,6 +1259,10 @@
    </property>
   </action>
   <action name="act_plugins_enable">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/system-turnon.svgz</normaloff>:/16x16/system-turnon.svgz</iconset>
+   </property>
    <property name="text">
     <string>Enable</string>
    </property>
@@ -1228,6 +1271,10 @@
    </property>
   </action>
   <action name="act_plugins_disable">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/system-shutdown.svgz</normaloff>:/16x16/system-shutdown.svgz</iconset>
+   </property>
    <property name="text">
     <string>Disable</string>
    </property>
@@ -1236,6 +1283,10 @@
    </property>
   </action>
   <action name="act_plugins_bypass">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/dry.svgz</normaloff>:/16x16/dry.svgz</iconset>
+   </property>
    <property name="text">
     <string>0% Wet (Bypass)</string>
    </property>
@@ -1244,6 +1295,10 @@
    </property>
   </action>
   <action name="act_plugins_wet100">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/wet.svgz</normaloff>:/16x16/wet.svgz</iconset>
+   </property>
    <property name="text">
     <string>100% Wet</string>
    </property>
@@ -1252,6 +1307,10 @@
    </property>
   </action>
   <action name="act_plugins_mute">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/audio-volume-muted.svgz</normaloff>:/16x16/audio-volume-muted.svgz</iconset>
+   </property>
    <property name="text">
     <string>0% Volume (Mute)</string>
    </property>
@@ -1260,6 +1319,10 @@
    </property>
   </action>
   <action name="act_plugins_volume100">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/audio-volume-medium.svgz</normaloff>:/16x16/audio-volume-medium.svgz</iconset>
+   </property>
    <property name="text">
     <string>100% Volume</string>
    </property>
@@ -1268,6 +1331,10 @@
    </property>
   </action>
   <action name="act_plugins_center">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/balance.svgz</normaloff>:/16x16/balance.svgz</iconset>
+   </property>
    <property name="text">
     <string>Center Balance</string>
    </property>
@@ -1435,6 +1502,17 @@
     <enum>QAction::NoRole</enum>
    </property>
   </action>
+  <action name="act_settings_show_toolbar_text">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>Show Toolbar Text</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
   <action name="act_settings_configure">
    <property name="icon">
     <iconset resource="../resources.qrc">
@@ -1553,7 +1631,23 @@
     <enum>QAction::NoRole</enum>
    </property>
   </action>
+  <action name="act_plugins_change_skin">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/skin.svgz</normaloff>:/16x16/skin.svgz</iconset>
+   </property>
+   <property name="text">
+    <string>Change &amp;Skin...</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
   <action name="act_plugins_compact">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/compact.svgz</normaloff>:/16x16/compact.svgz</iconset>
+   </property>
    <property name="text">
     <string>Compact Slots</string>
    </property>
@@ -1562,6 +1656,10 @@
    </property>
   </action>
   <action name="act_plugins_expand">
+   <property name="icon">
+    <iconset resource="../resources.qrc">
+     <normaloff>:/16x16/restore.svgz</normaloff>:/16x16/restore.svgz</iconset>
+   </property>
    <property name="text">
     <string>Expand Slots</string>
    </property>
@@ -1597,7 +1695,7 @@
   <action name="act_plugin_add_jack">
    <property name="icon">
     <iconset resource="../resources.qrc">
-     <normaloff>:/16x16/list-add.svgz</normaloff>:/16x16/list-add.svgz</iconset>
+     <normaloff>:/16x16/add-jack.svgz</normaloff>:/16x16/add-jack.svgz</iconset>
    </property>
    <property name="text">
     <string>Add &amp;JACK Application...</string>
diff -ur Carla/resources/ui/carla_settings.ui Carla.my17/resources/ui/carla_settings.ui
--- Carla/resources/ui/carla_settings.ui	2025-03-07 22:50:26.246329084 +0300
+++ Carla.my17/resources/ui/carla_settings.ui	2025-03-25 19:48:07.050968785 +0300
@@ -511,6 +511,27 @@
               </property>
              </spacer>
             </item>
+            <item row="2" column="0" colspan="3">
+             <layout class="QHBoxLayout" name="horizontalLayout_28">
+              <item>
+               <widget class="QLabel" name="label_main_skin_tweaks_1">
+                <property name="toolTip">
+                 <string>Example: 'Tweak' is for all skins; extra 'skinnameTweak' overrides it for that skin only.</string>
+                </property>
+                <property name="text">
+                 <string>Skin tweaks:</string>
+                </property>
+               </widget>
+              </item>
+              <item>
+               <widget class="QLineEdit" name="le_main_skin_tweaks">
+                <property name="toolTip">
+                 <string>Example: 'Tweak' is for all skins; extra 'skinnameTweak' overrides it for that skin only.</string>
+                </property>
+               </widget>
+              </item>
+             </layout>
+            </item>
            </layout>
           </widget>
          </item>
diff -ur Carla/resources/ui/xycontroller.ui Carla.my17/resources/ui/xycontroller.ui
--- Carla/resources/ui/xycontroller.ui	2025-03-07 22:50:26.246329084 +0300
+++ Carla.my17/resources/ui/xycontroller.ui	2025-03-25 19:48:07.051233979 +0300
@@ -30,41 +30,117 @@
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
-         <widget class="ScalableDial" name="dial_x">
-          <property name="minimum">
-           <number>-100</number>
+         <spacer name="verticalSpacer0">
+          <property name="orientation">
+           <enum>Qt::Vertical</enum>
           </property>
-          <property name="maximum">
-           <number>100</number>
+          <property name="sizeType">
+           <enum>QSizePolicy::Fixed</enum>
+          </property>
+          <property name="minimumSize">
+           <size>
+            <width>48</width>
+            <height>0</height>
+           </size>
+          </property>
+         </spacer>
+        </item>
+        <item>
+         <widget class="QWidget" name="dial_x">
+          <property name="minimumSize">
+           <size>
+            <width>48</width>
+            <height>58</height>
+           </size>
+          </property>
+          <property name="maximumSize">
+           <size>
+            <width>48</width>
+            <height>58</height>
+           </size>
+          </property>
+         </widget>
+        </item>
+        <item>
+         <widget class="QWidget" name="dial_out_x">
+          <property name="minimumSize">
+           <size>
+            <width>48</width>
+            <height>58</height>
+           </size>
+          </property>
+          <property name="maximumSize">
+           <size>
+            <width>48</width>
+            <height>58</height>
+           </size>
           </property>
          </widget>
         </item>
         <item>
-         <spacer name="verticalSpacer">
+         <spacer name="verticalSpacer1">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
-          <property name="sizeHint" stdset="0">
+          <property name="minimumSize">
            <size>
-            <width>20</width>
-            <height>30</height>
+            <width>48</width>
+            <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
-         <widget class="ScalableDial" name="dial_y">
-          <property name="minimum">
-           <number>-100</number>
+         <widget class="QWidget" name="dial_y">
+          <property name="minimumSize">
+           <size>
+            <width>48</width>
+            <height>58</height>
+           </size>
           </property>
-          <property name="maximum">
-           <number>100</number>
+          <property name="maximumSize">
+           <size>
+            <width>48</width>
+            <height>58</height>
+           </size>
+          </property>
+         </widget>
+        </item>
+        <item>
+         <widget class="QWidget" name="dial_out_y">
+          <property name="minimumSize">
+           <size>
+            <width>48</width>
+            <height>58</height>
+           </size>
+          </property>
+          <property name="maximumSize">
+           <size>
+            <width>48</width>
+            <height>58</height>
+           </size>
           </property>
          </widget>
         </item>
+        <item>
+         <spacer name="verticalSpacer2">
+          <property name="orientation">
+           <enum>Qt::Vertical</enum>
+          </property>
+          <property name="sizeType">
+           <enum>QSizePolicy::Fixed</enum>
+          </property>
+          <property name="minimumSize">
+           <size>
+            <width>48</width>
+            <height>0</height>
+           </size>
+          </property>
+         </spacer>
+        </item>
        </layout>
       </item>
      </layout>
